### PR TITLE
Reduce total occupancy

### PR DESCRIPTION
**Pull request by eringold**
Pull request overview
---------------------

Adjusts building occupancy schedules to closer align ComStock occupancy with US Commercial Building occupancy as determined from American Time Use Survey data.

[ComStock Change Document](https://nrel.sharepoint.com/:w:/r/sites/comstock/Shared%20Documents/ComStock%20Filing%20Cabinet/ComStock%20Core/Change%20Documentation/Changes/vTBD%20Occupancy/ComStock%20Update%20Total%20Occupancy.docx?d=wd2b82d92b392462ebdac9df0344f510a&csf=1&web=1&e=Qgk1pY)

### Pull Request Author

This pull request makes changes to (select all the apply):
 - [x] Workflow Measures

Author pull request checklist:

 - [x] Tagged the pull request with the appropriate label (documentation, infrastructure, sampling, workflow measure, upgrade measure, reporting measure, postprocessing) to help categorize changes in the release notes.
 - [x] Added tests for new measures
 - [x] Updated measure .xml(s)
 - [x] Both `options_lookup.tsv` files updated
 - [x] 10k+ test run
 - [x] Change documentation written
 - [x] ComStock documentation updated
 - [x] Added 'See ComStock License' language to first two lines of each code file

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [x] Perform a code review on GitHub
 - [x] All related changes have been implemented: data and method additions, changes, tests
 - [x] If fixing a defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [x] Reviewed change documentation
 - [x] Ensured code files contain License reference
 - [x] Results differences are reasonable
 - [x] CI status: all tests pass

#### ComStock Licensing Language - Add to Beginning of Each Code File
```
# ComStock™, Copyright (c) 2023 Alliance for Sustainable Energy, LLC. All rights reserved.
# See top level LICENSE.txt file for license terms.
```

### CODE REVIEW

File: resources/measures/adjust_occupancy_schedule/measure.rb [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    args = OpenStudio::Measure::OSArgumentVector.new
+
+    # make an argument for peak occupancy fraction
+    peak_occ_frac = OpenStudio::Measure::OSArgument.makeDoubleArgument('peak_occ_frac', true)
+    peak_occ_frac.setDisplayName('Peak Occupancy Fraction')
+    peak_occ_frac.setDefaultValue(0.6)
+    args << peak_occ_frac
+
+    # apply/not apply measure
+    apply_measure = OpenStudio::Ruleset::OSArgument.makeBoolArgument('apply_measure', true)
```
Comment (asparke2) [inline resources/measures/adjust_occupancy_schedule/measure.rb]: Why do we need an `apply_measure` argument? In ComStock, we would remove the `occupancy_schedule_adjust` input column from the buildstock.csv to take it out of the workflow. In other measures, this argument is vestigial from the days when ComStock was a PAT project and OS-server needed each measure to have at least one argument to turn on/off.

File: resources/measures/adjust_occupancy_schedule/resources/os_lib_schedules.rb [hunk 2]
```diff
@@ -0,0 +1,933 @@
+# *******************************************************************************
```
Comment (asparke2) [inline resources/measures/adjust_occupancy_schedule/resources/os_lib_schedules.rb]: Are the method used from this part of openstudio-standards yet? If so, use from there and remove resource file if possible.

**Review by asparke2 — CHANGES_REQUESTED** [review]
Minor code changes. Comment in the word doc RE calculating the % reduction fraction; did it account for the fact that ComStock only models ~66% of the floor area?

File: resources/measures/adjust_occupancy_schedule/measure.rb (hunk 1 above)
Comment (eringold) [reply resources/measures/adjust_occupancy_schedule/measure.rb]: I'll remove.

File: resources/measures/adjust_occupancy_schedule/resources/os_lib_schedules.rb (hunk 2 above)
Comment (eringold) [reply resources/measures/adjust_occupancy_schedule/resources/os_lib_schedules.rb]: I was [tracking](https://nrel.sharepoint.com/:w:/r/sites/comstock/Shared%20Documents/ComStock%20Filing%20Cabinet/ComStock%20Core/openstudio-standards%20refactor%20changes/openstudio-standards%20refactor%20change%20list.docx?d=w94e52e177b154a719fc3cae4e51fd6b4&csf=1&web=1&e=toT71v) this as a revision to make once the whole standards refactor is complete and released, but I think the methods used have already been implemented and could possibly be updated now.

File: resources/measures/adjust_occupancy_schedule/measure.rb [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    runner.registerInitialCondition("#{sch_names.size} Unique occupancy schedules found. Initial total nominal occupancy: #{tot_ppl_nom}, initial peak schedule-adjusted occupancy: #{tot_ppl_adj}.")
+
+    # loop through people schedules and apply adjustment
+    runner.registerInfo("Reducing occupancy schedule values by #{peak_occ_frac * 100}%. Design sizing schedule values will remain unchanged.")
+
+    sch_names.each do |h|
+      sch_name = h[:name]
+      schedule = model.getScheduleRulesetByName(sch_name).get
+      # append reduced percentage to schedule name
+      schedule.setName(sch_name + " #{peak_occ_frac * 100}%")
```
Comment (mdahlhausen) [inline resources/measures/adjust_occupancy_schedule/measure.rb]: may need a round or limit method so you don't make a schedule name like "my old schedule 23.11239394938400204324%"

File: resources/measures/adjust_occupancy_schedule/resources/os_lib_schedules.rb (hunk 2 above)
Comment (mdahlhausen) [reply resources/measures/adjust_occupancy_schedule/resources/os_lib_schedules.rb]: I agree with @asparke on this.
There may be an issue updating from this resource file to the new schedules methods.
I suggest copying these openstudio-standards schedule methods directly into the top of the measure.rb. It'll make it much easier to change them out in the refactor, and we won't have any surprises in them behaving differently:
```
- schedule_get_min_max
- schedule_ruleset_simple_value_adjust
```

File: resources/measures/adjust_occupancy_schedule/measure.rb (hunk 3 above)
Comment (mdahlhausen) [reply resources/measures/adjust_occupancy_schedule/measure.rb]: also make the name descriptive, like "reduced by" or "x% of sch"

**Comment by mdahlhausen** [comment]
Also add this measure to the workflow measure [README](https://github.com/NREL/ComStock/blob/main/resources/measures/README.md).

**Review by mdahlhausen — CHANGES_REQUESTED** [review]
The change documentation looks great - thanks for including all the comparisons!
A few minor changes:
- [x] Copy methods from openstudio-standards instead of using the methods in os_lib_schedules.rb. That will make it easier to refactor in the future.
- [x] Update the workflow measures [README](https://github.com/NREL/ComStock/blob/main/resources/measures/README.md) with this measure.
- [x] Add ComStock license to files.
- [x] Update ComStock documentation in the occupancy section to explain this update.

**Comment by eringold** [comment]
@asparke2 @mdahlhausen Thanks for the review. I think I've addressed all comments, let me know if there's anything else you'd like to see, particularly on the documentation.

**Review by mdahlhausen — APPROVED** [review]
Looks excellent!

**Review by mdahlhausen — APPROVED** [review]

**Review by asparke2 — APPROVED** [review]
Looks good, let's merge

**Pull request merged**